File: 7-reverse-integer/7-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        const int max_limit=INT_MAX;
    
    long long int ans=0;
    
    int k=abs(x);
    
    while(k>0){
        
        ans = ans*10 + (k%10);
        k /= 10;
        
    }
    
    if(ans > max_limit)
        return 0;
    
    if(x<0)
        return (-1)*ans;
    else
        return ans;

    }
};

File: 7-reverse-integer/7-reverse-integer.py
class Solution(object):
    def reverse(self, x):
		if x==0:
			return 0
		s=str(x).rstrip("0")
		s=s[::-1]
		if x<0:
			s="-"+s.rstrip("-")
		s=int(s)
		if (-2)**31>s or s>((2)**31)-1:
				return 0
		return s
        